code/backend/batch/utilities/helpers/embedders/PushEmbedder.py

```diff
@@ -17,39 +20,75 @@
-            self.processor_map[ext] = processor
+            self.embedding_configs[ext] = processor
 
     def embed_file(self, source_url: str, file_name: str):
         file_extension = file_name.split(".")[-1]
-        processor = self.processor_map.get(file_extension)
-        self.__embed(source_url=source_url, processor=processor)
+        embedding_config = self.embedding_configs.get(file_extension)
+        self.__embed(source_url=source_url, embedding_config=embedding_config)
         if file_extension != "url":
             self.blob_client.upsert_blob_metadata(
                 file_name, {"embeddings_added": "true"}
             )
 
-    def __embed(self, source_url: str, processor: EmbeddingConfig):
-        vector_store_helper = AzureSearchHelper()
-        vector_store = vector_store_helper.get_vector_store()
-        if not processor.use_advanced_image_processing:
-            try:
-                document_loading = DocumentLoading()
-                document_chunking = DocumentChunking()
-                documents: List[SourceDocument] = []
-                documents = document_loading.load(source_url, processor.loading)
-                documents = document_chunking.chunk(documents, processor.chunking)
-                keys = list(map(lambda x: x.id, documents))
-                documents = [
-                    document.convert_to_langchain_document() for document in documents
-                ]
-                return vector_store.add_documents(documents=documents, keys=keys)
-            except Exception as e:
-                logger.error(f"Error adding embeddings for {source_url}: {e}")
-                raise e
-        else:
-            logger.warn("Advanced image processing is not supported yet")
+    def __embed(self, source_url: str, embedding_config: EmbeddingConfig):
+        documents_to_upload: List[SourceDocument] = []
+        try:
+            if not embedding_config.use_advanced_image_processing:
+                documents: List[SourceDocument] = self.document_loading.load(
+                    source_url, embedding_config.loading
+                )
+                documents = self.document_chunking.chunk(
+                    documents, embedding_config.chunking
+                )
+
+                for document in documents:
+                    documents_to_upload.append(
+                        self._convert_to_search_document(document)
+                    )
+
+                response = (
+                    self.azure_search_helper.get_search_client().upload_documents(
+                        documents_to_upload
+                    )
+                )
+                if not all([r.succeeded for r in response]):
+                    raise Exception(response)
+
+            else:
+                logger.warning("Advanced image processing is not supported yet")
+
+        except Exception as e:
+            logger.error(f"Error adding embeddings for {source_url}: {e}")
+            raise e
```

Comment:
nitpick: same comment as earlier about error logging and throwing exception back.
I know there might be a lot of previous code like this and it is not fair to change everything in this PR. I would however suggest to follow boy scout's rule - leave the code better than you found it. If we touch something, let's keep cleaning that along the way.